packages/expo-application/src/Application.ts

```diff
@@ -114,17 +115,7 @@ export async function getIosIdForVendorAsync(): Promise<string | null> {
   if (!ExpoApplication.getIosIdForVendorAsync) {
     throw new UnavailabilityError('expo-application', 'getIosIdForVendorAsync');
   }
-  return (await ExpoApplication.getIosIdForVendorAsync()) ?? null;
-}
-
-// @docsMissing
-export enum ApplicationReleaseType {
-  UNKNOWN = 0,
-  SIMULATOR = 1,
-  ENTERPRISE = 2,
-  DEVELOPMENT = 3,
-  AD_HOC = 4,
-  APP_STORE = 5,
```

Comment:
Prefer the enum over type unions IMO (will defer to @tsapeta). If the issue is that the runtime values of the enums are human-unfriendly, we could make a breaking change to make them be strings (`enum { UNKNOWN = 'UNKNOWN', ... }`).